Sources/Amplitude/Utilities/HttpClient.swift

```diff
@@ -13,9 +13,10 @@ class HttpClient {
 
     init(configuration: Configuration) {
         self.configuration = configuration
-        // shared instance has limitations but think we are not affected
-        // https://developer.apple.com/documentation/foundation/urlsession/1409000-shared
-        self.session = URLSession.shared
+
+        let sessionConfiguration = URLSessionConfiguration.ephemeral
```

Comment:
One thing you may want to do here is to nil out the [`.urlCache`](https://developer.apple.com/documentation/foundation/urlsessionconfiguration/1410148-urlcache) property on the session configuration. This has the effect of disabling caching completely – rather than the default behaviour of the `.ephemeral` configuration which still creates an in-memory cache.

As this completely disables caching on the configuration, the distinction between `.ephemeral` and `.default` is nullified, so the complete specification of the configuration becomes:

```
let sessionConfiguration = URLSessionConfiguration.default
sessionConfiguration.urlCache = nil
sessionConfiguration.httpMaximumConnectionsPerHost = 2
```

Reply:
Thanks @tcldr. 

@falconandy I agree that we probably disable caching. I can't think of a case we would need it and could save the memory usage. wdyt?

Reply:
ok, updated to the suggested configuration